hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DefaultPortableReader.java

```diff
@@ -684,8 +687,11 @@ private PortablePosition findPositionForReading(String path) throws IOException
         }
     }
 
-    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:returncount"})
+    @SuppressWarnings({"checkstyle:cyclomaticcomplexity", "checkstyle:returncount", "unchecked"})
     private <T> T readSinglePositionFromNonArray(PortablePosition position) throws IOException {
+        if (position.getType() == null) {
+            throw new IllegalArgumentException("Unsupported type read: null");
+        }
         switch (position.getType()) {
```

Comment:
Why not an `assert`? I think they are appropriate everywhere except at the immediate point of contact with the user.